### DISCUSSION TITLE
How to calculate current home consumption power

### DISCUSSION THREAD

**Discussion by fahussain** [kellerza/sunsynk#281] [Q&A] [answered]
@kellerza thank you so much for this wonderful project. Your integration is giving us the control and information that would not have been possible with the manufacturer provided tools. 

I have setup my Inverex Nitrox 8kw (aka Deye 8kw) with home assistant using your integration with net metering. I am trying to setup a live view of the power consumption and I am struggling to figure it out. To be honest I am not sure how the following work:

1. Aux Power (I think it's called Smart Load for Deye)
2. Essentials (I think this is called backup Load for Deye)
3. Non Essentials (Load for Deye)
4. Grid Power
5. Grid CT Power
6. Inverter Power

Any information about the above would be extremely helpful.

Thanks

**Comment by fahussain** [comment, marked as answer]
I realized I was looking at it the wrong way. So the formula for current home power consumption turned out to be:
PV power(In my case it was pv1 and pv2) + Battery power + Grid power
The grid import/export and battery charge/discharge would be handled by + and - values.

**Reply by kellerza** [reply]
👍  the essentials and non-essentials sensors already do some of this math on the registers from the inverter: https://github.com/kellerza/sunsynk/blob/main/src/sunsynk/definitions/single_phase.py#L100

You could also use them as templates to create your own custom Mathsensor - see the docs on where to add these
